### PR TITLE
Remove dead code warnings from test

### PR DESCRIPTION
**Pull request by mfornet**
Use workaround proposed [here](https://github.com/rust-lang/rust/issues/46379#issuecomment-487421236) to avoid dead code warning.

### CODE REVIEW

**Comment by mfornet** [comment]
> Why do we have dead code in the first place?

tl;dr: We don't have dead code, but there is a "bug" in cargo.

It is a "bug" in cargo (or bad design). 
My use case was the following, I need to make some common tools for test, but didn't want to put them on `src/test_utils.rs` since it would make `network` depend on `client` (which is undesirable).
But adding it as dev-dependency is ok.

From rust book I found this [pattern](https://doc.rust-lang.org/book/ch11-03-test-organization.html#submodules-in-integration-tests) which was exactly what I needed. The problem with this is that every test file (under the `tests` folder) is seen by cargo as an independent crate, and since some tests don't use this common tools cargo complain about some dead code.

Read more about this issue on https://github.com/rust-lang/rust/issues/46379 (I also found the fix there).
